### PR TITLE
Make Node::BaseURI use the document's base URL

### LINKED ISSUE #11145: Make Node::BaseURI use the document's base URL
`Document::base_url()` exists, and it should be used here.

Specification: https://dom.spec.whatwg.org/#dom-node-baseuri
Code: `components/script/dom/node.rs`
Test: `./mach test-wpt tests/wpt/web-platform-tests/html/infrastructure/urls/terminology-0/document-base-url.html`

I would expect some test results to change, so they should be [updated](https://github.com/servo/servo/blob/master/tests/wpt/README.md#updating-test-expectations) accordingly.

### PR DESCRIPTION
**Pull request by mrmiywj**
Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data:
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #11145  (github issue number if applicable).

Either:
- [X] There are tests for these changes OR
- [ ] These changes do not require tests because _____

Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11178)

### CODE REVIEW

**Comment by nox** [comment]
@bors-servo try

r=me modulo the test expectation nit.

Thanks for your contribution @mrmiywj!

-S-awaiting-review +S-needs-code-changes 

<details><summary><i>Previously, highfive wrote…</i></summary>
> Thanks for the pull request, and welcome! The Servo team is excited to review your changes, and you should hear from @manishearth (or someone else) soon.</details>


---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[tests/wpt/metadata/html/infrastructure/urls/terminology-0/document-base-url.html.ini, line 4 \[r1\]](https://reviewable.io:443/reviews/servo/servo/11178#-KHfkEaFUT6q_4xIi8mY:-KHfkEaFUT6q_4xIi8mZ:1122823267) ([raw file](https://github.com/servo/servo/blob/e2ff29ba995f1082fa15cad0a24b901bc3e70fbd/tests/wpt/metadata/html/infrastructure/urls/terminology-0/document-base-url.html.ini#L4)):*
> ```INI
>   type: testharness
>   [The document base URL of a document containing one or more base elements with href attributes is the frozen base URL of the first base element in the document that has an href attribute, in tree order.]
>     expected: PASS
> ```

This whole entry should be removed, by default a test is considered to `PASS`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11178#-:-KHfkJNln2K41DzImV4V:-669084490)*

**Comment by nox** [comment]
@mrmiywj You should also update the expectations for `/html/infrastructure/urls/terminology-0/multiple-base.sub.html`.

**Comment by mrmiywj** [comment]
cc @nox 
Sorry for that I missed this test.

File: tests/wpt/metadata/html/infrastructure/urls/terminology-0/document-base-url.html.ini [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [document-base-url.html]
   type: testharness
   [The document base URL of a document containing one or more base elements with href attributes is the frozen base URL of the first base element in the document that has an href attribute, in tree order.]
-    expected: FAIL
+    expected: PASS
```
Comment (KiChjang) [inline tests/wpt/metadata/html/infrastructure/urls/terminology-0/document-base-url.html.ini]: You still haven't removed this entry.

File: tests/wpt/metadata/html/infrastructure/urls/terminology-0/multiple-base.sub.html.ini [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [multiple-base.sub.html]
   type: testharness
   [If there are multiple <base> elements, the document base URL is the frozen base URL of the first one that has an href attribute]
-    expected: FAIL
+    expected: PASS
```
Comment (KiChjang) [inline tests/wpt/metadata/html/infrastructure/urls/terminology-0/multiple-base.sub.html.ini]: This file can be deleted.

**Comment by mrmiywj** [comment]
cc @KiChjang 
Thx!

**Comment by KiChjang** [comment]
@bors-servo r=nox

Thanks!

**Pull request merged**